Teacher Workout/Features/Components/Form/InputFieldView.swift

```diff
@@ -0,0 +1,61 @@
+import SwiftUI
+
+struct InputFieldView: View {
+    var label: String
+    var iconName: String
+    var placeholder: String
+    var isSecureField: Bool = false
+    var errorMessage: String?
+
+    @Binding var fieldData: String
+    @Binding var showError: Bool
+
+    var body: some View {
+        VStack(alignment: .leading) {
+            Text(label)
+                .font(.caption)
+                .textCase(.uppercase)
+            HStack {
+                Image(systemName: iconName)
+                    .font(.system(size: 24))
+                    .foregroundColor(.secondary)
+
+                if isSecureField {
+                    SecureField(placeholder, text: $fieldData)
+                        .autocapitalization(.none)
+                } else {
+                    TextField(placeholder, text: $fieldData)
+                        .autocapitalization(.none)
+                        .keyboardType(.emailAddress)
+                }
+            }
+            .padding()
+            .overlay(
+                RoundedRectangle(cornerRadius: 8, style: .continuous)
+                    .stroke(showError ? Color.red : Color.secondary, lineWidth: 1)
+            )
+
+            if let error = errorMessage,
+               showError
+            {
```

Comment:
Braces should be on the same line with condition, to maintain the same coding style.